sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/SpendDialogPresenter.java

```diff
@@ -91,10 +91,10 @@ public void onResponse(OpenOrder response) {
 			public void onFailure(KinEcosystemException exception) {
 				showToast(SOMETHING_WENT_WRONG);
 				try {
-					String errorMsg = ((ApiException) exception.getCause()).getResponseBody().getMessage();
+					String errorMsg = exception.getMessage();
 					eventLogger.send(SpendOrderCreationFailed
 						.create(errorMsg, offer.getId(), false, SpendOrderCreationFailed.Origin.MARKETPLACE));
-				} catch (ClassCastException e) {
+				} catch (Throwable e) {
```

Comment:
no need to try catch both lines are identical and use exception.getMessage() as the message - this will also really help with catching crash as the exception.getMessage() is used again